### PR TITLE
Add Kata meta-package

### PR DESCRIPTION
**Pull request by miz060**
Add Kata meta-package

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Introduce kata meta-package for the UVM components.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- BuddyBuild pipeline: 
  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=471709&view=results
- [AMD64-3-OneBranch]-Dev-BuildImages: 
  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=472079&view=results
- 2.0 image build:
  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=472201&view=results
- performance test:
  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=472256&view=results
- conformance test: 
  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=472261&view=results

### CODE REVIEW

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
                 "crash-ptdump-command",
                 "delve",
                 "fstrm",
+                "kata-packages",
```
Comment (ms-mahuber) [inline SPECS/LICENSES-AND-NOTICES/data/licenses.json:15]: kata-packages-uvm?

File: .github/workflows/validate-cg-manifest.sh [hunk 2]
```diff
@@ -37,6 +37,7 @@ ignore_no_source_tarball=" \
   initramfs \
   installkernel \
   javapackages-tools-meta \
+  kata-packages \
```
Comment (ms-mahuber) [inline .github/workflows/validate-cg-manifest.sh:40]: kata-packages-uvm?

File: SPECS/kata-packages/kata-packages.spec [hunk 3]
```diff
@@ -0,0 +1,106 @@
+Summary:        Metapackage for Kata
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:1]: ... for Kata UVM components

File: SPECS/kata-packages/kata-packages.spec [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+Name:           kata-packages-uvm
+Version:        1.1.0
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:3]: I'd also start with 1.0.0 here

File: SPECS/kata-packages/kata-packages.spec [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+Release:        1%{?dist}
+License:        MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System Environment/Base
+URL:            https://aka.ms/mariner
+
+%description
+Metapackage for Kata.
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:12]: align w summary

File: SPECS/kata-packages/kata-packages.spec [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+%package        uvm
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:14]: this is the main package, not kata-packages-uvm

File: SPECS/kata-packages/kata-packages.spec [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+Requires:       sed
+Requires:       tar
+Requires:       tzdata
+Requires:       util-linux
+Requires:       zlib
+
+%description    uvm
+%{summary}
+
+%package        coco-uvm
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:44]: suggest to call this coco

File: SPECS/kata-packages/kata-packages.spec [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+Summary:        Metapackage to install the set of packages inside a Kata confidential containers UVM.
+Requires:       %{name}-uvm = %{version}-%{release}
+Requires:       device-mapper
+Requires:       opa
+
+%description    coco-uvm
+%{summary}
+
+%package        uvm-build
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:53]: call build

File: SPECS/kata-packages/kata-packages.spec [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+Requires:       python3-setuptools
+Requires:       python3-pytest 
+Requires:       python3-libclang
+Requires:       python3-tomli    
+Requires:       veritysetup 
+
+%description    uvm-build
+%{summary}
+
+%package        coco-uvm-sign
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:82]: call coco-sign

File: SPECS/kata-packages/kata-packages.spec [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+%files uvm
+
+%files coco-uvm
+
+%files uvm-build
+
+%files coco-uvm-sign
+
+%changelog
```
Comment (ms-mahuber) [inline SPECS/kata-packages/kata-packages.spec:104]: adapt if we start off of 1.0.0. Say we introduce the meta-package for the UVM components

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json (hunk 1 above)
Comment (miz060) [reply SPECS/LICENSES-AND-NOTICES/data/licenses.json]: Updated

File: .github/workflows/validate-cg-manifest.sh (hunk 2 above)
Comment (miz060) [reply .github/workflows/validate-cg-manifest.sh]: Updated.

File: SPECS/kata-packages/kata-packages.spec (hunk 3 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Update with suggestion.

File: SPECS/kata-packages/kata-packages.spec (hunk 5 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Updated.

File: SPECS/kata-packages/kata-packages.spec (hunk 7 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Updated.

File: SPECS/kata-packages/kata-packages.spec (hunk 8 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Updated.

File: SPECS/kata-packages/kata-packages.spec (hunk 10 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Updated.

File: SPECS/kata-packages/kata-packages.spec (hunk 9 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Updated.

File: SPECS/kata-packages/kata-packages.spec (hunk 4 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Updated.

File: SPECS/kata-packages/kata-packages.spec (hunk 6 above)
Comment (miz060) [reply SPECS/kata-packages/kata-packages.spec]: Updated.

File: .github/workflows/validate-cg-manifest.sh [hunk 11]
```diff
@@ -37,6 +37,7 @@ ignore_no_source_tarball=" \
   initramfs \
   installkernel \
   javapackages-tools-meta \
+  kata-packages-uvm \
```
Comment (ms-mahuber) [inline .github/workflows/validate-cg-manifest.sh:40]: minor question: do we need to list the subpackages here as well?

**Review by ms-mahuber — APPROVED** [review]

File: .github/workflows/validate-cg-manifest.sh (hunk 11 above)
Comment (christopherco) [reply .github/workflows/validate-cg-manifest.sh:40]: Entries in here are per .spec and not per-package

File: SPECS/kata-packages-uvm/kata-packages-uvm.spec [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+%files build
+
+%files coco-sign
+
+%changelog
+* Tue Dec 19 2023 Mitch Zhu <mitchzhu@microsoft.com> - 1.0.0-1
+- Introduce kata meta-package for the UVM components.
+- License verified
+- Original version for CBL-Mariner
```
Comment (christopherco) [inline SPECS/kata-packages-uvm/kata-packages-uvm.spec:102]: Add EOF newline

```suggestion
- Original version for CBL-Mariner

```

Comment (miz060) [reply SPECS/kata-packages-uvm/kata-packages-uvm.spec]: EOF newline added.

**Review by Redent0r — APPROVED** [review]

**Pull request merged**